propulsion-hotel/Reactor/Handler.fs

```diff
@@ -0,0 +1,88 @@
+module Reactor.Handler
+
+open Infrastructure
+open Propulsion.Internal
+
+[<RequireQualifiedAccess>]
+type Outcome = Completed | Deferred | Resolved of successfully : bool
+
+/// Gathers stats based on the outcome of each Span processed, periodically including them in the Sink summaries
+type Stats(log, statsInterval, stateInterval, ?logExternalStats) =
+    inherit Propulsion.Streams.Stats<Outcome>(log, statsInterval, stateInterval)
+
+    let mutable completed, deferred, failed, succeeded = 0, 0, 0, 0
+
+    override _.DumpStats() =
+        base.DumpStats()
+        if completed <> 0 || deferred <> 0 || failed <> 0 || succeeded <> 0 then
+            log.Information(" Completed {completed} Deferred {deferred} Failed {failed} Succeeded {succeeded}", completed, deferred, failed, succeeded)
+            completed <- 0; deferred <- 0; failed <- 0; succeeded <- 0
+        match logExternalStats with None -> () | Some f -> f Serilog.Log.Logger
```

Comment:
I'm not clear on why we'd pass the global logger to this function rather than the one passed to the constructor.

What context am I missing?

Reply:
Removed it
Have just realised what it was.... If you have 2 reactors in one app, then you might use a `.ForContext` to separate the messages.
These external stats don't 'belong to' any specific reactor, so they take a contextless log (which most of the wiring assumes there will be one of in other places).
Will reinstate with explaining comment

Reply:
Done